oneview/data_source_rack_manager.go

```diff
@@ -0,0 +1,185 @@
+// (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+package oneview
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceRackManager() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceRackManagerRead,
+		Schema: map[string]*schema.Schema{
+			"category": {
+				Type: schema.TypeString,
+
+				Optional: true,
+			},
+			"created": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"etag": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hostname": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"licensing_intent": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"location": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"model": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"modified": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"part_number": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"refresh_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"remote_support_uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"password": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"scopes_uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"serial_number": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"initial_scope_uris": {
+				Optional: true,
+				ForceNew: true,
```

Comment:
are we doing same in other cases? there was specific idempotency issues related to initial_scope_uri..hope we have take care of it here